slides/src/content/services.md

```diff
@@ -114,4 +114,134 @@ import { HttpModule } from '@angular/http';
-[View Example](https://plnkr.co/edit/E9xJ9nUGxOQ66AbOmGXL?p=preview)
\ No newline at end of file
+[View Example](https://plnkr.co/edit/E9xJ9nUGxOQ66AbOmGXL?p=preview)
+
+---
+
+## Handling `http` rejections
+
+To catch rejections we can use the `subscribe` operator's `onError` callback
+
+```ts
+  this.http.post(`${ BASE_URL }/auth/login`, payload)
+    .map(response => response.json())
+    .subscribe(
+      (authData) => this.storeToken(authData.id_token), // onNext function
+      (err) => console.error(err),                      // onError function
+      () => console.log('Authentication Complete')      // onComplete function
+    );
+```
+
+---
+
+## Using the `.catch` operator
+
+- Using `.catch` we can inspect an error and decide how to handle it
+- By not calling `subscribe` our code is more declarative
+- Can take advantage of the `async` pipe in your template
+
+```ts
+  search(term: string) {
+    return this.http.get(`https://api.spotify.com/v1/dsds?q=${term}&type=artist`)
+      .map(response => response.json())
+      .catch(e => {
+        if (e.status >==  500) return cachedVersion();
+
+        return Observable.throw(
+          new Error(`${ e.status } ${ e.statusText }`)
+        );
+      }
+  });
+```
+
+---
+
+## Cancel a Request
+
+- One of the greatest benefits to using observables over promises is the ability to cancel `http` requests
+
+```ts
+  search() {
+    const request = this.searchService.search(this.searchField.value)
+      .subscribe(
+        result => { this.result = result.artists.items; },
+        err => { this.errorMessage = err.message; },
+        () => { console.log('Completed'); }
+      );
+
+    setTimeout(() => {request.unsubscribe()}, 0); // cancel request after 0 milliseconds
+  }
+```
+[View Example](http://plnkr.co/edit/XQL8v9?p=preview)
+
+---
+
+## Cancel a Request with `takeUntil`
+
+- Managing subscription cancellation via `unsubscribe` can get messy
+- The `takeUntil` operator allows us to compose your subscription management
+- `takeUntil` emits items from an observable until a given observable emits an item
+
+```ts
+search() {
+  const request = this.searchService.search(this.searchField.value)
+    .takeUntil(someOtherStream) // take until someOtherStream emits an item
+    .subscribe(
+      result => { this.result = result.artists.items; },
+      err => { this.errorMessage = err.message; },
+      () => { this.cancelMessage = 'Your request has been cancelled' }
+    );
+}
+```
+
+- **NOTE:** `takeUntil` will complete the observable unlike `unsubscribe`
+
+[View Example](https://plnkr.co/edit/v2aGkTCmi34jBr7an1i8?p=preview)
+
+---
+
+## Retry a request
+
+- It is possible to retry a failed request using the `.retry` operator when something goes wrong, for example, the user's connection may dropped.
+- `retry` takes a `retryCount` argument that specifies the number of times to retry a failed request.
+- If you do not specify `retryCount`, the request will be retried indefinitely.
```

Comment:
We could be more concise in this section. This is my suggested change:

- Retry a failed request using the `retry` operator
- Useful when the user's connection fails
- `retry` takes an argument that specifies the number of retries
- If no argument is provided, the request will be retried indefinitely